### PR TITLE
chore: add simple test for addressables duplicates

### LINKED ISSUE #4366: Add validation to detect duplication in Addressables
#### Description

With moving more assets to addressable, we need to have a  validation mechanism to detect duplication resulting in build size and memory usage increase.
We should investigate adding a validation test to the CI, and fail when we detect asses being duplicated in the bundle.

We should have some filtering system to skip checks for some specific asset type, i.e shaders.
We should be able to target specific bundles for the validation, i.e only build-in assets.

For validation, we can use Addressables Editor API, as Unity uses for writing build-in analyze rules.

#### Acceptance Criteria
- [ ] Validation test added to the build pipeline
- [ ] CI failing if detect duplication in assets

### PR DESCRIPTION
**Pull request by popuz**
## What does this PR change?
Closes #4366

- Add validation tests, which are based on Addressables Analyze API (for more info on this decision read comments in the #4366).
- Tests are separated for Custom bundles, Resources and Scenes (same as in Addressables Analyze, re-use same Rules)
- Step in CI to run validation only for Custom bundles. Because Resources and Scenes bundles have too many duplicates. We will need to fix them first before re-enabling these tests
- Made automated fix for Duplicates in our bundles (but we will need to refactor this big group for better performance in future)
- Marked several failing Edit Mode tests as "Legacy" (Category), so we will be able to exclude them from the EditMode test Suite in future.
- Add EditModeCI Category to unlock possibility for EditMode tests in CI

## How to test the changes?

[Devs Only]
0. Open Editor -> TestRunner

1. Run AddressablesValidationTests
2. Validate that test for Custom Bundles is passing, but one for Resources and Scenes are failing.

1. Select only EditMode tests and choose Uncategorised in Category 
2. Run Tests and Validate that all 157 tests pass 

- Check in the CI artifacts that `editmode-results.xml` contains `ValidateDuplicateBundleDependencies` with "passed" status, similar to this:

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### CODE REVIEW

**Review by mikhail-dcl — APPROVED** [review]
Looks good to me

**Review by pbosio — APPROVED** [review]

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/AssetManager/Addressables/Tests/AddressablesEssentialsTests.cs [hunk 1]
```diff
@@ -9,76 +9,73 @@
...
+        var addressableProvider = new AddressableResourceProvider();
 
-            //Act
-            var dabAnim = await addressableProvider.GetAddressable<AnimationClip>("dab.anim");
-            var tikAnim = await addressableProvider.GetAddressable<AnimationClip>("tik.anim");
-            var loadingScreen = await addressableProvider.GetAddressable<GameObject>("_LoadingScreen");
+        //Act
+        var dabAnim = await addressableProvider.GetAddressable<AnimationClip>("dab.anim");
+        var tikAnim = await addressableProvider.GetAddressable<AnimationClip>("tik.anim");
+        var loadingScreen = await addressableProvider.GetAddressable<GameObject>("_LoadingScreen");
```
Comment (dalkia) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/AssetManager/Addressables/Tests/AddressablesEssentialsTests.cs:45]: Loading screen left the addressables. I think we can remove this one, since its going to fail

**Review by dalkia — APPROVED** [review]
